### PR TITLE
Housekeeping

### PR DESCRIPTION
**Pull request by nathan7**
We had a few loose bits of dead code, random things that didn't pass linting, and mixed code style throughout the codebase.
For starters, I fixed all of the nits. Everything passes JSHint, and there's no dead code hanging around.
To fix the mixed style, I propose using @feross's `standard`. It's quite close to your style, @ForbesLindesay, and ensures that both of us and any contributors will adhere to the repo style from now on. `standard` will run as part of our tests to ensure we stick to it.
I've reformatted the codebase to match, with some help from `standard --format`.

### CODE REVIEW

File: lib/core.js [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
-    Symbol() :
-    (
-      '_promise_internal_key_handle_' +
-      Math.random().toString(35).substr(2, 10) + '_'
-    )
+  Symbol() :
+  (
+  '_promise_internal_key_handle_' +
+  Math.random().toString(35).substr(2, 10) + '_'
+  )
```
Comment (ForbesLindesay) [inline lib/core.js]: This is easier to read with the additional indentation for the bracketed expression.  Readability > Lintability

**Comment by ForbesLindesay** [comment]
I don't want us to use "standard" for two reasons:

1. There are a few points I disagree on (e.g. I prefer `function foo() {` rather than `function foo () {`) but these are not a deal breaker.
2. "standard" is in the business of actively misleading newcomers to JavaScript and node.js.  This is ultimately a deal breaker.  If a coding style is to be called "standard" it must aim to codify the status quo, rather than define its own style.  In particular, the JavaScript community uses semi-colons in the vast majority of codebases.  When I wrote this library I was experimenting with semi-colon free, but have since come round to the idea that using semi-colons is just easier and more natural.

There are lots of other nits I would happily take (e.g. space after `//`), but I'm reluctant to make linting part of the tests (if it's checking for things like whitespace rather than just looking for undeclared globals) because it ultimately slows down people attempting to just write code, and I would very rarely reject a pull request that improved the library based purely on poor placement of whitespace.

**Comment by feross** [comment]
@ForbesLindsay It's no problem if you don't want to use `standard` - it's not for everyone :-)

I initially shared your concerns about slowing down people attempting to just write code. Like you, I would rarely ever reject a pull request purely on poor whitespace, or even more serious errors. Instead, I always merged and fixed the issues in a followup commit. I think this is a much nicer experience for newbie contributors and turns more of them into regular contributors :-)

At the same time, setting up clear, automated contributor expectations makes a project healthier, in my experience. Nearly everyone who is working on a PR is already running `npm test` and would happily fix up style (and sometimes more serious issues) that `standard` catches. I've already fixed half-a-dozen bugs related to forgetting to handle the `err` parameter in node-style callbacks and other serious errors, because `standard` caught them.

Lastly, I'm sorry that you think `standard` is actively misleading JavaScript newbies. That was never my intention - I was just trying to make it easier for newbies to contribute to my own modules and didn't expect it so many people to share out the link to it. I've added language to the readme to clarify that `standard` is not a real "web standard" or affiliated with any official standards groups. It's just one standard, among many, that you can choose to follow.

**Comment by feross** [comment]
Also, if "no semicolons" is the only rule you object to, you might consider using https://npmjs.com/package/semistandard by @Flet.

**Comment by ForbesLindesay** [comment]
That may be an option, although this library is mostly semi-colon free, so that would require a substantial change.

The issue is that the warning about it not being a standard is buried at the bottom of the readme, but the marketing hype that strongly implies that it **is** an accepted standard runs from the name, all the way through most of the rest of the readme.

It would be great to have a project exactly like standard, that codifies all the most popular decisions of the JavaScript community (e.g. for every eslint option, use static analysis to scrape GitHub and turn the lint rule on if less than half of all JavaScript files fail that lint rule test).  I agree that these issues are generally not worth bikesheding over, but that's exactly what "standard" encourages by choosing a highly contentious standpoint with certain issues.

I think there's a strong case for a tool that purely catches "likely errors" such as not handling the err argument of a node callback, touching global variables other than the whitelisted browser and node.js globals etc. but does not ever report bugs with whitespace/semicolons.  A tool like that would always be a no-brainer to add to any project.

**Pull request closed without merging**